manifests/charts/istio-control/istio-discovery/files/gateway-injection-template.yaml

```diff
@@ -131,10 +131,15 @@ spec:
       timeoutSeconds: 3
       failureThreshold: {{ .Values.global.proxy.readinessFailureThreshold }}
     volumeMounts:
+    - name: workload-socket
+      mountPath: /var/run/secrets/workload-spiffe-uds
     {{- if eq .Values.global.caName "GkeWorkloadCertificate" }}
-    - name: gke-workload-certificate
+    - name: workload-certs
       mountPath: /var/run/secrets/workload-spiffe-credentials
       readOnly: true
```

Comment:
nit: I think we can probably drop the conditional and merge these, even though it means it will not be readOnly... @lei-tang any concerns?

Reply:
>     1. In original code, the CSI driver volume "workloadcertificates.security.cloud.google.com" is created if global.caName is "GkeWorkloadCertificate". Why is "gke-workload-certificate" changed to "workload-certs"?
> 

As it's the same path we thought it makes sense to use the same volume name. I'll revert that change it you prefer to keep "gke-workload-certificate" when the caName is GkeWorkloadCertificate.
 
>     2. When caName is not "GkeWorkloadCertificate", why does the PR mount an emptyDir to /var/run/secrets/workload-spiffe-credentials which is used to mount GKE workload certificate? Which CA will provision cert files to this emptyDir? How to configure Istio to choose this CA as the CA provider?
> 

The idea was to reuse the same path for mounting certificates for the the Workload. There's no configuration needed, it detects the presence of the certs files on the path.

Reply:
To be consistent with the caName "GkeWorkloadCertificate" and to avoid growing this PR into changing the existing GkeWorkloadCertificate workflow, I think it may be better to keep the original code for the caName "GkeWorkloadCertificate" as is (such as "gke-workload-certificate"). What do you think?
> 
> The idea was to reuse the same path for mounting certificates for the the Workload. There's no configuration needed, it detects the presence of the certs files on the path.
Which entity will provision the cert files on the path?

Reply:
I am not sure there is much value in keeping the GKE-x specific names. The whole idea is that the GKE cert mounting is no longer a GKE specific thing, but a standard that is implemented by multiple vendors

Reply:
Yes, I think it can be like this:

```
volumes:
  {{- if eq .Values.global.caName "GkeWorkloadCertificate" }}
  - name: gke-workload-certificate
    csi:
      driver: workloadcertificates.security.cloud.google.com
  {{- else}}
  - emptyDir:
    name: workload-certs
  {{- end }}

volumeMounts:
    {{- if eq .Values.global.caName "GkeWorkloadCertificate" }}
    - name: gke-workload-certificate
      mountPath: /var/run/secrets/workload-spiffe-credentials
      readOnly: true
    {{- else }}
    - name: workload-certs
      mountPath: /var/run/secrets/workload-spiffe-credentials
    {{- end }}
```
We still need to put the `workload-certs` on the `else` because we are reusing the same path.

Reply:
@maxlambrecht The proposed code looks good to me. Thanks!